browser/brave_wallet/cardano_provider_impl_unittest.cc

```diff
@@ -0,0 +1,526 @@
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using testing::_;
+
+namespace brave_wallet {
+
+namespace {
+
+class MockBraveWalletService : public BraveWalletService {
+ public:
+  MockBraveWalletService() {}
+  ~MockBraveWalletService() override {}
+
+  MOCK_METHOD0(keyring_service, KeyringService*());
+};
+
+class MockBraveWalletProviderDelegate : public BraveWalletProviderDelegate {
+ public:
+  MockBraveWalletProviderDelegate() = default;
+  ~MockBraveWalletProviderDelegate() override {}
+
+  MOCK_METHOD0(IsTabVisible, bool());
+  MOCK_METHOD0(ShowPanel, void());
+  MOCK_METHOD0(ShowWalletBackup, void());
+  MOCK_METHOD0(UnlockWallet, void());
+  MOCK_METHOD0(WalletInteractionDetected, void());
+  MOCK_METHOD0(ShowWalletOnboarding, void());
+  MOCK_METHOD1(ShowAccountCreation, void(mojom::CoinType type));
+  MOCK_CONST_METHOD0(GetOrigin, url::Origin());
+  MOCK_METHOD3(RequestPermissions,
+               void(mojom::CoinType type,
+                    const std::vector<std::string>& accounts,
+                    RequestPermissionsCallback));
+  MOCK_METHOD2(IsAccountAllowed,
+               bool(mojom::CoinType type, const std::string& account));
+  MOCK_METHOD2(GetAllowedAccounts,
+               std::optional<std::vector<std::string>>(
+                   mojom::CoinType type,
+                   const std::vector<std::string>& accounts));
+  MOCK_METHOD1(IsPermissionDenied, bool(mojom::CoinType type));
+  MOCK_METHOD1(AddSolanaConnectedAccount, void(const std::string& account));
+  MOCK_METHOD1(RemoveSolanaConnectedAccount, void(const std::string& account));
+  MOCK_METHOD1(IsSolanaAccountConnected, bool(const std::string& account));
+};
+
+}  // namespace
+
+class CardanoProviderImplUnitTest : public testing::Test {
+ public:
+  CardanoProviderImplUnitTest()
+      : shared_url_loader_factory_(
+            base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+                &url_loader_factory_)) {}
+  ~CardanoProviderImplUnitTest() override = default;
+
+  void SetUp() override {
+    RegisterLocalStatePrefs(local_state_.registry());
+    RegisterProfilePrefs(prefs_.registry());
+    RegisterProfilePrefsForMigration(prefs_.registry());
+    network_manager_ = std::make_unique<NetworkManager>(&prefs_);
+    json_rpc_service_ = std::make_unique<JsonRpcService>(
+        shared_url_loader_factory_, network_manager_.get(), &prefs_, nullptr);
+    keyring_service_ = std::make_unique<KeyringService>(json_rpc_service_.get(),
+                                                        &prefs_, &local_state_);
+    brave_wallet_service_ = std::make_unique<MockBraveWalletService>();
+    ON_CALL(*brave_wallet_service_, keyring_service())
+        .WillByDefault(
+            ::testing::Invoke([&]() { return keyring_service_.get(); }));
+    provider_ = std::make_unique<CardanoProviderImpl>(
```

Comment:
would default behavior of `brave_wallet_service_->keyring_service()` be enought?